In [1]:
import pandas as pd
from sklearn.linear_model import LogisticRegression
from sklearn.ensemble import RandomForestClassifier
from xgboost import XGBClassifier
from sklearn.metrics import accuracy_score, f1_score, roc_auc_score

X_train = pd.read_csv('/content/X_train.csv')
X_test = pd.read_csv('/content/X_test.csv')

def evaluate_classifier(y_train_path, y_test_path, model, model_name, task_name):
    y_train = pd.read_csv(y_train_path).values.ravel()
    y_test = pd.read_csv(y_test_path).values.ravel()
    model.fit(X_train, y_train)
    y_pred = model.predict(X_test)
    y_proba = model.predict_proba(X_test)[:, 1]
    print(f"\n{task_name} - {model_name}")
    print(f"  Accuracy: {accuracy_score(y_test, y_pred):.3f}")
    print(f"  F1-score: {f1_score(y_test, y_pred):.3f}")
    print(f"  ROC-AUC:  {roc_auc_score(y_test, y_proba):.3f}")

print("="*60)
print("КЛАССИФИКАЦИЯ SI > 8")
print("="*60)

evaluate_classifier('y_train_si_8.csv', 'y_test_si_8.csv',
    LogisticRegression(max_iter=1000, random_state=42),
    "LogisticRegression", "SI > 8")

evaluate_classifier('y_train_si_8.csv', 'y_test_si_8.csv',
    RandomForestClassifier(n_estimators=100, random_state=42),
    "RandomForest", "SI > 8")

evaluate_classifier('y_train_si_8.csv', 'y_test_si_8.csv',
    XGBClassifier(n_estimators=100, random_state=42, eval_metric='logloss'),
    "XGBoost", "SI > 8")

КЛАССИФИКАЦИЯ SI > 8

SI > 8 - LogisticRegression
  Accuracy: 0.649
  F1-score: 0.000
  ROC-AUC:  0.537

SI > 8 - RandomForest
  Accuracy: 0.682
  F1-score: 0.503
  ROC-AUC:  0.693

SI > 8 - XGBoost
  Accuracy: 0.707
  F1-score: 0.542
  ROC-AUC:  0.696


Вывод:классификация SI > 8 оказалась сложной задачей. Наилучший результат показал XGBoost (AUC = 0.696, Accuracy = 70.7%), однако низкий F1-score (0.542) свидетельствует о том, что модель часто пропускает высокоселективные соединения. Логистическая регрессия полностью несостоятельна. Полученные модели могут использоваться лишь для первичной фильтрации, но не для окончательного отбора кандидатов. Для улучшения качества требуется дополнительный отбор признаков или использование более сложных архитектур.